ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -273,17 +273,20 @@ public KsqlPlan plan(
   @Override
   public ExecuteResult execute(final ServiceContext serviceContext, final ConfiguredKsqlPlan plan) {
     try {
+      if (!plan.getPlan().isPresent()) {
+        return ExecuteResult.of("should not execute");
```

Comment:
Is there another message we can use instead of `should not execute`? Btw, is this a bug that execute is called with an empty plan?

Reply:
Refactored this out